### ISSUE TITLE
Acton compilation & package structure

### ISSUE THREAD

**Issue by plajjan** [actonlang/acton#44]
All the compilation output currently goes in the same directory, which is in the Acton system directory. After #43 we at least split up so that system builtin and shipped modules are in `libActon.a` whereas project related go in `libActonProject.a`, still, both are placed in the `lib` directory in the system directory.

We should split this up to make it possible to install Acton on a system outside of a users home directory, yet let the user compile an Acton project in their home directory.

Acton projects typically live in the users home directory and when compiled should produce output in the Acton project directory. This also makes it possible to have multiple Acton projects with potentially overlapping module names but without creating conflicts.

Today, an `.acton` file marks a directory as a module? Is this correct? It's a little weird to me - I think they're too far in, I think I would rather mark something  an Acton project and then presume the remainder of the directory structure based on that, something along the lines of (where Acton.toml marks this as an Acton project, similar to Rust's Cargo which has a Cargo.toml):
```
my-acton-project/
├── Acton.toml
├── out
│   ├── bin
│   │   └── main
│   ├── lib
│   └── modules
└── src
    ├── foo
    │   └── bar.act
    └── main.act
```

in `main.act` we can then do
```
import foo.bar
```
and access the things. I'm not quite sure how to structure the compiled output, but I think it should go in my-acton-project at least. I put a `out` directory that more or less replicates the things we currently write to the acton system directory but in the project directory instead. Under it we can write a project local modules and lib. Did a POC patch (there is some debug output printed first, ignore... or not). It finds the project path based on the presence of a `Acton.toml` file there, then assumes `src` exists in there etc:
```
kll@ThinkYoga:~/kod/my-acton-project$ actonc --verbose src/foo/kaka/banan.act 
projPath: /home/kll/kod/my-acton-project
subdirs: src foo kaka
dirInSrc: foo kaka
fileBody: banan
srcPath: /home/kll/kod/my-acton-project/src/foo/kaka
## sysPath  : /home/kll/terastream/acton
## sysMod   : /home/kll/terastream/acton/modules
## sysLib   : /home/kll/terastream/acton/lib
## sysRoot  : /home/kll/terastream/acton/modules
## projPath : /home/kll/kod/my-acton-project
## projMod  : /home/kll/kod/my-acton-project/out/modules
## projLib  : /home/kll/kod/my-acton-project/out/lib
## srcRoot  : /home/kll/kod/my-acton-project/src
## modPrefix: foo
## topMod   : foo.kaka.banan
Compiling /home/kll/kod/my-acton-project/src/kaka/banan.act... Done.
kll@ThinkYoga:~/kod/my-acton-project$ 
```
In this case the Acton system is "installed" in `/home/kll/terastream/acton` (which we "find" based on the location of actonc, which I've included in my `PATH`). This should really be somewhere in `/usr` for a proper installed system - it is somewhere to which only root can write and not my user. Thus, we cannot write any output there. The builtin modules etc are in there though!

I found `sysRoot` confusing, like I have managed to learn that it is a subdirectory to `sysPath` (I sort of want to think the root is higher up) but then I can never remember whether `sysRoot` is the root of the lib files or the code (.c / .h / .ty)... so I propose we replace it with `sysMod` which is clearer, at least to me. Similarly we have `sysLib` for the lib directory. Since #43 we already put project library output in a separate libActonProject.a and libActon.a contains the system builtin / included libraries. Thus, for this we just need to change so we put libActonProject.a in `projLib`. When building we need to add (`-L`) both the `projLib` and `sysLib` directories to ensure we can find `libActonProject.a` and `libActon.a`. Cgen output goes in `projMod` instead of `sysRoot` / `sysMod` like now. We obviously need to include both when building so we can find the right `.h` and `.ty` files.

I'm not entirely sure about the meaning of topMod and modPrefix but I think I wrangled them right in my POC.
# Changes
- rename srcRoot to projPath
  - new projPath variable should have same value as old srcRoot
  - there will be a new srcRoot
- add srcRoot variable = `projPath ++ "/src"`
- rename `--path` argument to `--syspath` to further clarify what it is for
- make `sysPath` default to a compile time value which depends on where we intend to install something for a particular platform
- new `sysLib` is `sysPath ++ /lib`
- new `sysMod` is `sysPath ++ /modules`
- new `projLib` is `projPath ++ /lib`
- new `projMod` is `projPath ++ /modules`

## Acton system structure
I wrote above that the structure of the Acton system should be defined at compile time, like we define sysPath. On a Debian (like most Linuxes) this would prolly be `/usr/lib/acton`, so if we compute `sysLib` and `sysMod` based on that, we'd end up with like:
```
/usr/bin/actonc
/usr/lib/acton/lib/
/usr/lib/acton/modules/
```

It's a little weird to have a `lib` directory nested under our under `/usr/lib/acton` but they're sort of different libs, like this isn't the "compile any C library on your system with these lib files", so our lib is different. Yet, I do think the `acton` dir belongs in `/usr/lib`. There are many things in there that aren't C libraries we can link to and it seems our stuff belongs there. Alternative is like `/usr/share` but there's mostly other stuff in there?

I explicitly want to say that I don't think it's an option to place our libs in `/usr/lib` and our generated C code (.h header files)  in `/usr/include`. If you only were to look at the type of file, like "this is a .h file" you might be tempted to put it in `/usr/include` but our files are not for generic C use - they are specific to Acton and thus do not belong there.

On FreeBSD I think the equivalent is:
```
/usr/local/bin/actonc
/usr/local/lib/acton/lib/
/usr/local/lib/acton/modules/
```

We notice that the paths relative to each other and `actonc` are actually the same, so it might be that we could keep the current approach of determining sysPath based on the location of `actonc` - just that we also assume a structure, specifically:
```
bin/actonc
lib/acton/lib/
lib/acton/modules/
```
so like
- `sysPath = joinPath [ takeDirectory $ takeDirectory execDir, "lib", "acton" ]`
- `sysLib = joinPath [ sysPath, "lib" ]
- `sysMod = joinPath [ sysPath, "modules" ]

we could make this work in the acton source repo just by moving files around a bit, then it'd always be the same! Simple!? Or we just supply `--syspath` when invoked in the development repo...

**Comment by plajjan** [comment]
I pushed my hackity hack POC concept: https://github.com/actonlang/acton/commit/1a9f1e6f2badb9fd95c0b440c21020ef6998faf3 / https://github.com/actonlang/acton/tree/44-actonc-dir-poc

@nordlander have a read and let's discuss this at some point :)
